internal/sbi/consumer/ue_context_management.go

```diff
@@ -0,0 +1,98 @@
+	"github.com/free5gc/smf/internal/logger"
+	"github.com/free5gc/smf/internal/util"
+	"github.com/pkg/errors"
+)
+
+func UeCmRegistration(smCtx *smf_context.SMContext) (
+	*models.ProblemDetails, error,
+) {
+	uecmUri := util.SearchNFServiceUri(smf_context.GetSelf().UDMProfile, models.ServiceName_NUDM_UECM, models.NfServiceStatus_REGISTERED)
+	if uecmUri == "" {
+		return nil, errors.Errorf("SMF can not select an UDM by NRF: SearchNFServiceUri failed")
+	}
+
+	configuration := Nudm_UEContextManagement.NewConfiguration()
+	configuration.SetBasePath(uecmUri)
+	client := Nudm_UEContextManagement.NewAPIClient(configuration)
+
+	registrationData := models.SmfRegistration{
+		SmfInstanceId:               smf_context.GetSelf().NfInstanceID,
+		SupportedFeatures:           "",
+		PduSessionId:                smCtx.PduSessionId,
+		SingleNssai:                 smCtx.SNssai,
+		Dnn:                         smCtx.Dnn,
+		EmergencyServices:           false,
+		PcscfRestorationCallbackUri: "",
+		PlmnId:                      smCtx.Guami.PlmnId,
+		PgwFqdn:                     "",
+	}
+
+	logger.PduSessLog.Infoln("UECM Registration SmfInstanceId:", registrationData.SmfInstanceId, " PduSessionId:", registrationData.PduSessionId,
+		" SNssai:", registrationData.SingleNssai, " Dnn:", registrationData.Dnn, " PlmnId:", registrationData.PlmnId)
+
+	_, httpResp, localErr := client.SMFRegistrationApi.SmfRegistrationsPduSessionId(context.Background(),
+		smCtx.Supi, smCtx.PduSessionId, registrationData)
+	defer func() {
+		if httpResp != nil {
+			if rspCloseErr := httpResp.Body.Close(); rspCloseErr != nil {
+				logger.PduSessLog.Errorf("UeCmRegistration response body cannot close: %+v",
+					rspCloseErr)
+			}
+		}
+	}()
+
+	if localErr == nil {
+		smCtx.UeCmRegistered = true
+		return nil, nil
+	} else if httpResp != nil {
+		if httpResp.Status != localErr.Error() {
+			return nil, localErr
+		}
+		problem := localErr.(openapi.GenericOpenAPIError).Model().(models.ProblemDetails)
+		return &problem, nil
+	} else {
+		return nil, openapi.ReportError("server no response")
+	}
+}
+
+func UeCmDeregistration(smCtx *smf_context.SMContext) (*models.ProblemDetails, error) {
+	uecmUri := util.SearchNFServiceUri(smf_context.GetSelf().UDMProfile, models.ServiceName_NUDM_UECM, models.NfServiceStatus_REGISTERED)
+	if uecmUri == "" {
+		return nil, errors.Errorf("SMF can not select an UDM by NRF: SearchNFServiceUri failed")
+	}
+
+	configuration := Nudm_UEContextManagement.NewConfiguration()
+	configuration.SetBasePath(uecmUri)
+	client := Nudm_UEContextManagement.NewAPIClient(configuration)
+
+	httpResp, localErr := client.SMFDeregistrationApi.Deregistration(context.Background(),
+		smCtx.Supi, smCtx.PduSessionId)
+	defer func() {
+		if httpResp != nil {
+			if rspCloseErr := httpResp.Body.Close(); rspCloseErr != nil {
+				logger.ConsumerLog.Errorf("UeCmDeregistration response body cannot close: %+v",
+					rspCloseErr)
+			}
+		}
+	}()
+	if localErr == nil {
```

Comment:
```go
smCtx.UeCmRegistered = false
```

Reply:
Thanks, As suggested, updated the state in the UeCmDeregistration()